lib/blacklight/request_builders.rb

```diff
@@ -56,7 +56,7 @@ def search_builder_class
     end
 
     def search_builder processor_chain = search_params_logic
-      search_builder_class.new(processor_chain, self)
+      @search_builder ||= search_builder_class.new(processor_chain, self)
```

Comment:
I gave an explanation on #1110, so we can pass things to the search_builder (i.e. current_ability) without overriding `search_builder`